internal/pkg/manager/spod/spod_controller.go

```diff
@@ -275,6 +280,22 @@ func (r *ReconcileSPOd) handleCreate(
 		}
 	}
 
+	hasServiceMonitorCRD, err := r.hasServiceMonitorCRD(ctx)
+	if err != nil {
+		return errors.Wrap(err, "check for ServiceMonitor CRD")
+	}
+
+	if hasServiceMonitorCRD {
+		r.log.Info("Deploying operator service monitor")
+		if err := r.client.Create(
```

Comment:
you could just do the call to `Create` and if the error indicates that the object is not registered or is on an unknown type, then you could just ignore it.

e.g.

```
if runtime.IsNotRegisteredError(err) || meta.IsNoMatchError(err) {
```

where the functions come from:

```
	"k8s.io/apimachinery/pkg/runtime"
	"k8s.io/apimachinery/pkg/api/meta"
```

Reply:
Ohhh, that’s a good workaround. I’ll change it.